README.md

```diff
@@ -1,64 +1,119 @@
+A simple CLI templating tool written in golang.
 
 [![GitHub version](https://badge.fury.io/gh/noqcks%2Fgucci.svg)](https://badge.fury.io/gh/noqcks%2Fgucci)
 [![License](https://img.shields.io/github/license/noqcks/gucci.svg)](https://github.com/noqcks/gucci/blob/master/LICENSE)
 [![Build Status](https://travis-ci.org/noqcks/gucci.svg?branch=master)](https://travis-ci.org/noqcks/gucci)
 
-# Get
+## Installation
 
-If you have go installed
+If you have `go` installed:
 
 ```
-go get github.com/noqcks/gucci
+$ go get github.com/noqcks/gucci
 ```
 
-Or you can just download the binary and move it into your path
+Or you can just download the binary and move it into your `PATH`:
 
 ```
-VERSION=0.0.4
+VERSION=0.1.0
 wget -q https://github.com/noqcks/gucci/releases/download/v${VERSION}/gucci-v${VERSION}-darwin-amd64
 chmod +x gucci-v${VERSION}-darwin-amd64
 mv gucci-v${VERSION}-darwin-amd64 /usr/local/bin/gucci
 ```
 
+## Use
+
+### Locating Templates
 
-# Use
+`gucci` can locate a template in multiple ways.
 
-gucci can take input in multiple ways
+#### File
 
-### file
+Pass the template file path as the first argument:
 
 ```
-$ gucci template.tpl > template.conf
+$ gucci template.tpl > template.out
 ```
 
-### stdin
+#### Stdin
 
-```
+Supply the template through standard input:
+
+```bash
 $ gucci
 Start typing stuff {{ print "here" }}
 ^d
 Start typing stuff here
 ```
 
-via piping
+Via piping:
 
-```
+```bash
 $ echo '{{ html "<escape-me/>" }}' | gucci
 ```
 
-# Templating
+### Supplying Variable Inputs
+
+`gucci` can receive variables for use in templates in the following ways (in order of lowest to highest precedence):
+* A JSON or YAML file
+* Environment variables
+* Variable command options
+
+#### Variables File
+
+Given an example variables file:
+```yaml
+# vars.yaml
+hosts:
+  - name: bastion
+  - name: app
+```
+
+Pass it into `gucci` with `-f` or `--vars-file`:
+```bash
+$ gucci -f vars.yaml template.tpl
+```
+
+#### Environment Variables
+
+Here, `MY_HOST` is available to the template:
+```bash
+$ export MY_HOST=localhost
+$ gucci template.tpl
+```
+
+#### Variable Options
+
+Pass variable options into `gucci` with `-s` or `--set-var`, which can be repeated:
+```bash
+$ gucci -s foo.bar=baz template.tpl
+```
 
-### GoLang Functions
+Variable option keys are split on the `.` character, and nested such that
+the above example would equate to the following yaml variable input:
+
+```yaml
+foo:
+  bar: baz
+```
+
+## Templating
+
+#### GoLang Functions
 
 All of the existing [golang templating functions](https://golang.org/pkg/text/template/#hdr-Functions) are available for use.
 
-### Built In Functions
+#### Built In Functions
 
 This is a list of custom functions this tool adds that you can use:
 
+- `join`: Join a list of strings
```

Comment:
I was testing this out but it doesn't seem to be working. 

`./gucci -f vars.yml template.tpl`
```
Error occurred while attempting to template:
Failed to parse standard input: template: template.tpl:5:8: executing "template.tpl" at <.mir>: wrong type for value; expected []string; got []interface {}
```
The values.yml 
```
MIRRORS:
  - something
  - something1
  - something2
```
The template.tpl uses the example. 

I think it should work with this, but maybe im unsure of how this would look in the wild.

Reply:
Interesting. I'll see if I can reproduce that.

Reply:
Yep, the problem is that the `join` func needs a little more intelligence to convert `[]interface{}` into `[]string`. Since adding `join` was just a "hey.. why not join along with split?" thought, and not necessary for the main purpose of this PR, I've removed it. I can take a better shot at it in the future.